library/Graphite/Graphing/MetricsQuery.php

```diff
@@ -128,13 +132,25 @@ public function fetchColumn()
                 continue;
             }
 
-            $workaroundMacro = str_replace('.', '_', $macro);
-            if ($workaroundMacro === 'service_name') {
-                $workaroundMacro = 'service_description';
+            if ($this->object instanceof Model) {
+                // icingadb macros
+                $workaroundMacro = $macro;
+                if ($workaroundMacro === 'service.check_command') {
+                    $workaroundMacro = 'service.checkcommand';
+                }
+                if ($workaroundMacro === 'host.check_command') {
+                    $workaroundMacro = 'host.checkcommand';
+                }
+                $result = $this->resolveMacro($workaroundMacro, $this->object);
```

Comment:
I fear that there may be templates out there, which use even more macros that only the monitoring module understands. Have you checked all the default templates for any macros that need a migration? Please also think of asking our colleagues in the #tech channel for their experiences regarding macros in templates.